app/src/store/utils.ts

```diff
@@ -17,3 +60,39 @@ export const setInitialStateWithLSTableOptions = <T>(
     tableOptions,
   };
 };
+
+export const getUrlSearchParam = (value: string) => {
+  if ('URLSearchParams' in window) {
+    const searchParams = new URLSearchParams(window.location.search);
+
+    const param = searchParams.get(value);
+
+    return param;
+  }
+};
+
+export const setUrlSearchParams = (key: string, value: string | number) => {
+  if ('URLSearchParams' in window) {
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.set(key, typeof value === 'number' ? value.toString() : value);
+
+    const newRelativePathQuery = `${
+      window.location.pathname
+    }?${searchParams.toString()}`;
+
+    // eslint-disable-next-line no-restricted-globals
+    history.pushState(null, '', newRelativePathQuery);
+  }
+};
+
+export const setTableOptionsUrlSearchParams = (tableOptions: TableOptions) => {
```

Comment:
Just going to reply here for all the comments regarding naming in this file.  When I was reading this code I found myself looking at these where they are used a state changing functions.  So it confused me in a few places -- Maybe let's just leave everything how it is right now and discuss further as a team if we think this will be an issue or not